src/metabase/lib/js.cljs

```diff
@@ -556,6 +556,30 @@
   [a-query]
   (lib.core/TemplateTags-> (lib.core/template-tags a-query)))
 
+(defn ^:export does-require-native-collection
+  "Returns whether this database requires a native colletion."
+  [database-id metadata]
+  (lib.core/requires-native-collection? (metadataProvider database-id metadata)))
+
+(defn ^:export with-native-collection
+  "Changes the db collection to run this query against.
+   The first stage must be a native type. The database must support `native-specified-collection`"
+  [a-query collection-name]
+  (lib.core/with-native-collection a-query collection-name))
+
+(defn ^:export native-collection
+  "Returns the db collection associated with this query."
+  [a-query]
+  (lib.core/native-collection a-query))
```

Comment:
Whenever we encounter a similar new property, do we implement three such functions? (This might be good enough, because we might not encounter any similar property in the next ten years, but it feels a bit strange.)